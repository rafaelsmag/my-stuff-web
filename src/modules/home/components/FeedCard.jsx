import React, { Component } from 'react'
import CardFeed from '../../shared/components/Card'
import CardHeader from './CardHeader'
import CardBody from './CardBody'

class FeedCard extends Component {
  render () {
    const { post } = this.props
    return (
      <CardFeed>
        <CardHeader title={post.title} tags={post.tags} author={post.user}/>
        <CardBody images={post.images} text={post.body} />
      </CardFeed>
    )
  }
}

export default FeedCard
