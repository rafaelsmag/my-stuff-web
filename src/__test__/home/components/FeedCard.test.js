import React from 'react'
import { shallow } from 'enzyme'

import FeedCard from '../../../modules/home/components/FeedCard'

const props = {
  post: {
    _id: '1',
    title: 'Lorem ipsum dolor sit amet',
    body: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
    images: ['https://static.pexels.com/photos/371633/pexels-photo-371633.jpeg'],
    user: {
      image: 'https://www.shareicon.net/data/128x128/2016/07/26/802026_man_512x512.png',
      name: 'Steve Sanders'
    },
    tags: [{_id: '1', name: 'Sobremesas'}, {_id: '2', name: 'Bolo'}]
  }
}

describe('FeedCard Component', () => {
  const feedCard = shallow(<FeedCard {...props} />)
  it('should redenders properly', () => {
    expect(feedCard).toMatchSnapshot()
  })
})
