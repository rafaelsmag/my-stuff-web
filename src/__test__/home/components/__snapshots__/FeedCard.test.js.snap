// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedCard Component should redenders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FeedCard
    post={
        Object {
            "_id": "1",
            "body": "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
            "images": Array [
              "https://static.pexels.com/photos/371633/pexels-photo-371633.jpeg",
            ],
            "tags": Array [
              Object {
                "_id": "1",
                "name": "Sobremesas",
              },
              Object {
                "_id": "2",
                "name": "Bolo",
              },
            ],
            "title": "Lorem ipsum dolor sit amet",
            "user": Object {
              "image": "https://www.shareicon.net/data/128x128/2016/07/26/802026_man_512x512.png",
              "name": "Steve Sanders",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <CardHeader
          author={
                    Object {
                              "image": "https://www.shareicon.net/data/128x128/2016/07/26/802026_man_512x512.png",
                              "name": "Steve Sanders",
                            }
          }
          onClick={[Function]}
          tags={
                    Array [
                              Object {
                                "_id": "1",
                                "name": "Sobremesas",
                              },
                              Object {
                                "_id": "2",
                                "name": "Bolo",
                              },
                            ]
          }
          title="Lorem ipsum dolor sit amet"
/>,
        <CardBody
          images={
                    Array [
                              "https://static.pexels.com/photos/371633/pexels-photo-371633.jpeg",
                            ]
          }
          text="Lorem ipsum dolor sit amet, consectetur adipiscing elit"
/>,
      ],
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "author": Object {
            "image": "https://www.shareicon.net/data/128x128/2016/07/26/802026_man_512x512.png",
            "name": "Steve Sanders",
          },
          "onClick": [Function],
          "tags": Array [
            Object {
              "_id": "1",
              "name": "Sobremesas",
            },
            Object {
              "_id": "2",
              "name": "Bolo",
            },
          ],
          "title": "Lorem ipsum dolor sit amet",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "images": Array [
            "https://static.pexels.com/photos/371633/pexels-photo-371633.jpeg",
          ],
          "text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <CardHeader
            author={
                        Object {
                                    "image": "https://www.shareicon.net/data/128x128/2016/07/26/802026_man_512x512.png",
                                    "name": "Steve Sanders",
                                  }
            }
            onClick={[Function]}
            tags={
                        Array [
                                    Object {
                                      "_id": "1",
                                      "name": "Sobremesas",
                                    },
                                    Object {
                                      "_id": "2",
                                      "name": "Bolo",
                                    },
                                  ]
            }
            title="Lorem ipsum dolor sit amet"
/>,
          <CardBody
            images={
                        Array [
                                    "https://static.pexels.com/photos/371633/pexels-photo-371633.jpeg",
                                  ]
            }
            text="Lorem ipsum dolor sit amet, consectetur adipiscing elit"
/>,
        ],
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "author": Object {
              "image": "https://www.shareicon.net/data/128x128/2016/07/26/802026_man_512x512.png",
              "name": "Steve Sanders",
            },
            "onClick": [Function],
            "tags": Array [
              Object {
                "_id": "1",
                "name": "Sobremesas",
              },
              Object {
                "_id": "2",
                "name": "Bolo",
              },
            ],
            "title": "Lorem ipsum dolor sit amet",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "images": Array [
              "https://static.pexels.com/photos/371633/pexels-photo-371633.jpeg",
            ],
            "text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
